include/boost/astronomy/time/time_conversions.hpp

```diff
@@ -32,13 +29,70 @@ using namespace boost::posix_time;
  * The Greenwich Sidereal Time(GST) is the sidereal time correct for observations
  * made on the Greenwich meridian, longitude 0◦.
  */
-decimal_hour GST(ptime t)
+
+double Julian_date(boost::posix_time::ptime t)
 {
     //Get date from UT
-    date d = t.date();
+    boost::gregorian::date dt = t.date();
+
+    //Set y = year, m = month and d = day
+    double y = dt.year();
+    double m = dt.month();
+    double d = dt.day();
+
+    //If m = 1 or 2, set yprime = y − 1 and mprime = m + 12; otherwise yprime = y and mprime = m.
+    double yprime;
+    double mprime;
+    if ( (m == 1) || (m == 2) )
+    {
+        yprime = y - 1;
+        mprime = m + 12;
+    }
+    else
+    {
+        yprime = y;
+        mprime = m;
+    }
+
+    //Calculate B, check if the date is later than 1582 October 15
+    double B;
+    boost::gregorian::date dt1(1582, boost::gregorian::Oct , 1);
+    if ( dt > dt1 )
+    {
+        double A = floor(yprime / 100);
+        B = 2 - A + floor(A / 4);
+    }
+    else
+    {
+        B = 0;
+    }
+
+    //Calculate C, check if yprime is negative
+    double C;
+    if (yprime < 0)
+    {
+        C = floor((365.25 * yprime) - 0.75);
+    }
+    else
+    {
+        C = floor(365.25 * yprime);
+    }
 
+    //Calculate D
+    double D;
+    D = floor(30.6001 * (mprime + 1));
+
+    //Finding Julian date
+    double JD;
+    JD = B + C + D + d + 1720994.5;
+
+    return JD;
+}
+
+decimal_hour GST(boost::posix_time::ptime t)
+{
     //Get Julian Day Number
-    double JD = d.julian_day(); //Ambiguity in Julian precision.
+    double JD = Julian_date(t);
```

Comment:
We follow snake_case conventions in general kindly make the changes  

`julian_date`